### PR TITLE
Add 'ffmpeg_timeout' flag to write_frames

### PR DESCRIPTION
**Pull request by djhoese**
Possible solution/workaround for #13. This PR adds the 'ffmpeg_timeout' keyword argument to `write_frames` so the user can ask for imageio to wait longer for ffmpeg to finish. This is especially important when writing larger frames where ffmpeg may need more time.

### CODE REVIEW

**Comment by djhoese** [comment]
Could `invoke checkformat`'s output be any less helpful? I suppose it could leave out what file is the problem. Took me forever as someone not familiar with `invoke` to realize it was running black and that there was a `autoformat` task that I could have run (`invoke autoformat`).

File: tests/test_io.py [hunk 1]
```diff
@@ -253,6 +253,32 @@ def test_write_macro_block_size():
...
+            test_file2, (2048, 2048), pix_fmt_in=pixfmt, ffmpeg_timeout=20.0
+        )
+        gen.send(None)  # seed
+        for i in range(9):
+            data = (255 * np.random.rand(2048 * 2048 * bpp)).astype(np.uint8)
+            data = bytes(data)
+            gen.send(data)
+        gen.close()
+        with open(test_file2, "rb") as f:
+            sizes.append(len(f.read()))
```
Comment (almarklein) [inline tests/test_io.py]: Is seems like sizes is not used.

File: tests/test_io.py [hunk 2]
```diff
@@ -253,6 +253,32 @@ def test_write_macro_block_size():
...
+        # Check nframes
+        nframes, nsecs = imageio_ffmpeg.count_frames_and_secs(test_file2)
+        assert nframes == 9
```
Comment (almarklein) [inline tests/test_io.py]: Can you confirm that using the default 5s timeout will make this test fail? Maybe it's worth trying to set the timeout to 5, commit, push, see if CI fails?

**Comment by almarklein** [comment]
This looks like a great workaround. I will have also have a try at the solution using smaller buffers, but this PR is worthwhile nonetheless. I shall also have a look at making the invoke checks more informative :)

File: tests/test_io.py (hunk 1 above)
Comment (djhoese) [reply tests/test_io.py]: This test was mostly copied from other tests by @jmetz and existing tests like `test_write_pix_fmt_in` create `sizes` but don't use it. Would you like it removed or a check added?

File: tests/test_io.py (hunk 2 above)
Comment (djhoese) [reply tests/test_io.py]: I could try calling it twice with the first one having the default timeout. I was also thinking about setting it to 1s to avoid a case where ffmpeg gets so fast that it can complete this faster than 5s. Let me try adding an expected fail first.

File: tests/test_io.py (hunk 1 above)
Comment (jmetz) [reply tests/test_io.py]: Ya sorry - was just quick copy-edit job to create a test that caused the process killing issue. Hadn't fully audited the code.

File: tests/test_io.py (hunk 2 above)
Comment (jmetz) [reply tests/test_io.py]: As far as I can tell 5s should work... though it will be machine-dependent. What is the rationale for not going for a higher value? In the case of an actual hang of some kind, does it matter if the program waits 5 or 10s for example?

File: tests/test_io.py (hunk 1 above)
Comment (djhoese) [reply tests/test_io.py]: Doing the size check shows some odd results, at least from what I'd expect. The 'rgb24' file is larger than the 'rgba' file. Maybe has to do with it removing certain alpha=0 pixels?

File: tests/test_io.py (hunk 2 above)
Comment (almarklein) [reply tests/test_io.py]: > What is the rationale for not going for a higher value? In the case of an actual hang of some kind, does it matter if the program waits 5 or 10s for example?

Good point. I support it's mostly that I don't know what a good timeout would be. Should 10s be enough? or 20? Or maybe 50 in some cases? But if between us we've seen that 10s is enough, let's set it to 10 :)  I suspect that using a smaller buffer will lower the required timeout, but probably there will still be edge-cases, so this PR is worthwhile nonetheless.

File: tests/test_io.py (hunk 1 above)
Comment (almarklein) [reply tests/test_io.py]: Maybe ffmpeg uses different  compression ratios for rgb24 and rgba?

File: tests/test_io.py (hunk 2 above)
Comment (djhoese) [reply tests/test_io.py]: So as I'm preparing to test this, should this raise an exception instead of "silently" continuing on? If `ffmpeg` didn't close graciously, can't we assume it didn't create the file properly?

Comment (djhoese) [reply tests/test_io.py]: ...or does it fail with an exception? I feel like I've seen both.

Comment (djhoese) [reply tests/test_io.py]: Another option would be to do `warnings.warn` instead of `logger.warning` so that it is easier to catch in a test.

Comment (almarklein) [reply tests/test_io.py]: I would expect the assertion to fail, right? But maybe it's not that reproducible (i.e. sometimes ffmpeg might just be quick enough)?

Comment (djhoese) [reply tests/test_io.py]: @almarklein Which assertion? Right now, the writing of frames only logs a warning if ffmpeg times out. If you try to read/use the video file it created then it is corrupt and not useable. That's why in my new version of the test I check for the ffmpeg process to fail when it *reads* the file. It never outwardly fails (warnings.warn, exception) for the write.

Comment (almarklein) [reply tests/test_io.py]: I meant the `assert nframes == 9`, but I see what you mean now. Sounds good :)

Comment (djhoese) [reply tests/test_io.py]: Unless you want me to change that log warning to a warnings.warn or exception, I think this PR is ready for merge based on everything else that was discussed. Let me know if you want anything else changed.

**Comment by almarklein** [comment]
Thanks @djhoese for this fix, and everyone else who chimed in to help fix this!

**Comment by almarklein** [comment]
I made some further tweaks in #15. In short: changing `bufsize` did not affect the wait time, so I increased the default `ffmpeg_timeout` to 20s.
